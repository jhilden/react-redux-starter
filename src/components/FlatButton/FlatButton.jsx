import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Router, Route, Link } from 'react-router';
import classNames from 'classnames';

import Styles from './FlatButton.sass';

/**
 * FlatButton
 */
export default class FlatButton extends Component {
  handleOnClick = ::this.handleOnClick;
  handleMouseEnter = ::this.handleMouseEnter;
  handleMouseLeave = ::this.handleMouseLeave;

  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func.isRequired
  }

  state = {
    class: ''
  }

  handleOnClick(event) {
    this.props.onClick(event);
  }

  handleMouseEnter() {
    this.setState({class: 'is-hovered'});
  }

  handleMouseLeave() {
    this.setState({class: ''});
  }

  render() {
    return (
      <div
        className={classNames(Styles.FlatButton, this.state.class)}
        onClick={this.handleOnClick}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        {this.props.children}
      </div>
    )
  }
}
