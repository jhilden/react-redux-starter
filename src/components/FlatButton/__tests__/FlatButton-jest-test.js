import React from 'react';
import FlatButton from '../FlatButton';
import renderer from 'react-test-renderer';

test('FlatButton changes the class when clicked', () => {
  const component = renderer.create(
    <FlatButton onClick={function(event) { return; }}>Click</FlatButton>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});