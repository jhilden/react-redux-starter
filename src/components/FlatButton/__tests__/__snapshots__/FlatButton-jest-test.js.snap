exports[`test FlatButton changes the class when clicked 1`] = `
<div
  className="FlatButton"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Click
</div>
`;

exports[`test FlatButton changes the class when clicked 2`] = `
<div
  className="FlatButton is-hovered"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Click
</div>
`;
